tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/findandreplace/TestFindReplaceLogic.java

```diff
@@ -0,0 +1,275 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.findandreplace;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.After;
+import org.junit.Test;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.TextViewer;
+
+public class TestFindReplaceLogic {
+
+	Shell parentShell;
+	IFindReplaceLogic findReplaceLogic;
+	TextViewer textViewer;
+
+	public void setupFindReplaceLogicObject() {
+		setupFindReplaceLogicObject(null);
+	}
+
+	public void setupFindReplaceLogicObject(TextViewer target) {
+		findReplaceLogic= new FindReplaceLogic();
+		if (target != null) {
+			findReplaceLogic.updateTarget(target.getFindReplaceTarget(), true);
+		}
+	}
+
+	public void setupTextViewer(String contentText) {
+		parentShell= new Shell();
+		textViewer= new TextViewer(parentShell, SWT.BORDER | SWT.V_SCROLL | SWT.H_SCROLL);
+		textViewer.setDocument(new Document(contentText));
+		textViewer.getControl().setFocus();
+	}
+
+	@After
+	public void tearDownObjects() {
+		if (textViewer != null) {
+			textViewer.getControl().dispose();
+			textViewer= null;
+		}
+		parentShell.dispose();
+	}
+
+	@SuppressWarnings("boxing")
+	private void expectStatusIs(FindReplaceLogicMessageStatus test, FindReplaceLogicMessageStatus expected) {
+		assertEquals("Status message is not the expected message", test.getMessage(), expected.getMessage());
+		assertEquals("Status warning flag is not the expected value", test.isWarning(), expected.isWarning());
+		assertEquals("Status error flag is not the expected value", test.isError(), expected.isError());
+		findReplaceLogic.resetStatus(); // sketchy side-effect of this function.
+	}
+
+	private final String baseReplaceAllTestCaseSetupString= "aaaa";
```

Comment:
Either put this into a static final variable at the top of the class or simple inserts the string in the according methods, as its actual content has not specific meaning.